eks/cleanup.go

```diff
@@ -270,18 +270,25 @@ func waitForNetworkInterfacesToBeDetached(
 			niResult, err := ec2Svc.DescribeNetworkInterfaceAttribute(describeNetworkInterfacesInput)
 
 			switch {
-			// If there's no error, we have to keep trying.
+			// Yay, we're detached, process the next network interface.
+			case err == nil && isNIDetached(niResult):
+				logger.Infof("Network interface %s is detached.", aws.StringValue(ni.NetworkInterfaceId))
+				return nil // exit retry loop with success
+
+			// Since we checked whether the NI was detached in the first case, no error in this switch means the NI is
+			// not detached, so we need to retry.
 			case err == nil:
 				if niResult.Attachment != nil {
 					logger.Warnf("Network interface %s attachment status: %s", aws.StringValue(ni.NetworkInterfaceId), aws.StringValue(niResult.Attachment.Status))
 				}
 				return errors.WithStackTrace(fmt.Errorf("Network Interface %s not detached.", aws.StringValue(ni.NetworkInterfaceId))) // continue retrying
 
-			// Yay, we're detached, process the next network interface.
-			case isNIDetachedErr(niResult, err) || isNINotFoundErr(err):
-				logger.Infof("Network interface %s is detached.", aws.StringValue(ni.NetworkInterfaceId))
+			// If the NI can not be found, then it is already deleted so halt the loop and move on to the next NI.
```

Comment:
NIT: cannot is also a word. 😬 

```suggestion
			// If the NI cannot be found, then it is already deleted so halt the loop and move on to the next NI.
```

Reply:
TIL!